bluesky/bundlers.py

```diff
@@ -555,63 +555,74 @@ async def collect(self, msg):
             Msg('collect', flyer_object, stream=True, return_payload=False)
         """
-        obj = msg.obj
+        collect_obj = msg.obj
 
         if not self.run_is_open:
             # sanity check -- 'kickoff' should catch this and make this
             # code path impossible
             raise IllegalMessageSequence(
                 "A 'collect' message was sent but no " "run is open."
             )
-        self._uncollected.discard(obj)
+        self._uncollected.discard(collect_obj)
 
-        if hasattr(obj, "collect_asset_docs"):
+        if hasattr(collect_obj, "collect_asset_docs"):
             # Resource and Datum documents
-            for name, doc in obj.collect_asset_docs():
+            for name, doc in collect_obj.collect_asset_docs():
                 # Add a 'run_start' field to the resource document on its way out.
                 if name == "resource":
                     doc["run_start"] = self._run_start_uid
                 await self.emit(DocumentNames(name), doc)
 
-        named_data_keys = obj.describe_collect()
-        # e.g., {name_for_desc1: data_keys_for_desc1,
-        #        name for_desc2: data_keys_for_desc2, ...}
+        collect_obj_configuration = {}
+        if hasattr(collect_obj, "read_configuration"):
+            doc_logger.debug("reading configuration from %s", collect_obj)
+            collect_obj_configuration = collect_obj.read_configuration()
+        else:
+            doc_logger.debug("%s has no read_configuration method", collect_obj)
+
         bulk_data = {}
-        local_descriptors = {}  # hashed on obj_read, not (name, objs_read)
-        for stream_name, data_keys in named_data_keys.items():
-            desc_key = stream_name
-            d_objs = frozenset(data_keys)
-            if desc_key not in self._descriptors:
+        local_descriptors = {}  # hashed on objs_read, not (name, objs_read)
+        # collect_obj.describe_collect() returns a dictionary like this:
+        #     {name_for_desc1: data_keys_for_desc1,
+        #      name_for_desc2: data_keys_for_desc2, ...}
+        for stream_name, stream_data_keys in collect_obj.describe_collect().items():
+            d_objs = frozenset(stream_data_keys)
+            if stream_name not in self._descriptors:
                 objs_read = d_objs
-                # We don't not have an Event Descriptor for this set.
+                # We do not have an Event Descriptor for this set.
                 descriptor_uid = new_uid()
-                object_keys = {obj.name: list(data_keys)}
+                stream_object_keys = {collect_obj.name: list(stream_data_keys)}
                 hints = {}
-                if hasattr(obj, "hints"):
-                    hints.update({obj.name: obj.hints})
+                # is this if block an error?
+                if hasattr(collect_obj, "hints"):
+                    hints.update({collect_obj.name: collect_obj.hints})
+                for obj_read in objs_read:
```

Comment:
I think this loop is misconceived and should be removed. The `objs_read` is a `frozenset` of _strings_ represent data keys. The strings will never have a `.name` attribute or a `.hints` attribute. It's worth reviewing `_save` and `_monitor` to see if the same mistake was made there as well.

Reply:
It looks like the save method has
```python
for obj in objs_read:
    ...
    if hasattr(obj, "hints"):
        hints[name] = obj.hints
```
but monitor does not.

Reply:
removing
```python
    if hasattr(obj, "hints"):
        hints[name] = obj.hints
```
causes this [test](https://github.com/bluesky/bluesky/blob/master/bluesky/tests/test_run_engine.py#L1390) to fail. Is the test correct?

Reply:
Looks like `save` should be left as is. In that case, `self._obj_read` is a `frozenset` of actual objects. See https://github.com/bluesky/bluesky/pull/1338/files#r412169748